src/OSPSuite.Core/Domain/UnitSystem/DimensionFactory.cs

```diff
@@ -24,18 +29,37 @@ public void AddDimension(IDimension dimension)
          _dimensions.Add(dimension.Name, dimension);
       }
 
-      public IEnumerable<IDimension> Dimensions => _dimensions;
-
-      public IEnumerable<string> DimensionNames => _dimensions.Keys;
-
       public IDimension Dimension(string name)
       {
          if (!Has(name))
-            throw new KeyNotFoundException("Dimension " + name + " not available in DimensionFactory.");
+            throw new KeyNotFoundException(unknownDimension(name));
 
          return _dimensions[name];
       }
 
+      public bool TryGetDimension(string dimensionName, out IDimension dimension)
```

Comment:
THis is slightly new. Here we return the dimension if it exists by name
If not, this is potentially something like `Hydraulic conductivity per time` for which we have no dimension. THis was created automatically by MoBi. What we do is that we try to get it based on the name. If we succeed, all good, we return this one. AT that moment, the RHS Dimension was added to the lsit of dimension and is available during life time of the project